pkg/apis/pxc/v1/pxc_types.go

```diff
@@ -235,10 +236,10 @@ type ServerVersion struct {
 }
 
 type App interface {
-	AppContainer(spec *PodSpec, secrets string) corev1.Container
+	AppContainer(spec *PodSpec, secrets string, cr *PerconaXtraDBCluster) (corev1.Container, error)
 	SidecarContainers(spec *PodSpec, secrets string) []corev1.Container
-	PMMContainer(spec *PMMSpec, secrets string, v120OrGreater bool) corev1.Container
-	Volumes(podSpec *PodSpec) *Volume
+	PMMContainer(spec *PMMSpec, secrets string, cr *PerconaXtraDBCluster) (corev1.Container, error)
+	Volumes(podSpec *PodSpec, cr *PerconaXtraDBCluster) (*Volume, error)
 	Resources(spec *PodResources) (corev1.ResourceRequirements, error)
```

Comment:
I saw some "resources logic" in `AppContainer`methods. It should either here or there. If need it in the AppContainer, so maybe we no longer need Resources methods?